tests/bluetooth/shell/testcase.yaml

```diff
@@ -32,15 +32,17 @@ tests:
 # Bluetooth Audio Compile validation tests
   bluetooth.shell.no_vcs:
     build_only: true
-    filter: dt_label_with_parent_compat_enabled("storage_partition", "fixed-partitions")
+    filter: dt_label_with_parent_compat_enabled("storage_partition", "fixed-partitions") and
+            CONFIG_UART_INTERRUPT_DRIVEN
```

Comment:
This feels like a reversed way of doing it. 

If something needs some configuration to be enabled, the test should be guarded with that. Adding a filter for each other test case feels like a massive amount of work that should in a different way. 

I am curious as why this filter was only added for the test here that disables specific Bluetooth GATT services, but no in the above tests.

Reply:
Not sure which specific test you are referring to, but several of the above tests are limiting themselves via `platform_allow` and thus saying the test is only valid on the "allow" platforms.